src/test/java/com/coveo/pushapiclient/ApiUrlTest.java

```diff
@@ -0,0 +1,94 @@
+package com.coveo.pushapiclient;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class ApiUrlTest {
+
+    private ApiUrl defaultUrl;
+    private ApiUrl regionOnlyUrl;
+    private ApiUrl regionOnlyUrlCaseInsensitive;
+    private ApiUrl environmentOnlyUrl;
+    private ApiUrl environmentAndRegionUrl;
+    private ApiUrl streamURL;
+
+    @Before
+    public void setUp() throws MalformedURLException {
+        defaultUrl = new ApiUrl(
+                new URL("https://api.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/documents"));
+        regionOnlyUrl = new ApiUrl(
+                new URL("https://api-au.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/documents"));
+        regionOnlyUrlCaseInsensitive = new ApiUrl(
+                new URL("https://api-EU.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/documents"));
+        environmentOnlyUrl = new ApiUrl(
+                new URL("https://apidev.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/documents"));
+        environmentAndRegionUrl = new ApiUrl(
+                new URL("https://apidev-au.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/documents"));
+        streamURL = new ApiUrl(
+                new URL("https://apidev-au.cloud.coveo.com/push/v1/organizations/my-org-id/sources/my-source-id/stream/open"));
```

Comment:
I feel this one should have it's own test case because it is using the stream url? It's cool to see we're accepting `/open` and working with that,